### PR TITLE
Firebase Installations: Instance ID now uses FIS under the hood.

### PR DESCRIPTION
**Pull request by maksymmalyhin**
- default FirebaseApp FID is used as the app identity in FirebaseInstanceID
- not required FirebaseInstanceID code to manage the app identity removed
- `./scripts/pod_lib_lint.rb.`: a parameter `--ignore-local-specs` to accommodate FirebaseInstallations tests requiring a particular version of FirebaseInstanceID

### CODE REVIEW

File: scripts/pod_lib_lint.rb [hunk 1]
```diff
@@ -42,13 +48,28 @@ def main(args)
 
   command = %w(bundle exec pod lib lint --sources=https://cdn.cocoapods.org/)
 
+  # Split arguments that need to be processed by the script itself and passed
+  # to the pod command.
+  script_args = {}
```
Comment (maksymmalyhin) [inline scripts/pod_lib_lint.rb]: @wilhuff I invited you to provide the feedback on the `pod_lib_lint.rb` and `.travis.yaml` changes.

**Comment by paulb777** [comment]
Run `pod deintegrate`

**Comment by paulb777** [comment]
The Firebase.xcodeproj is broken. 

I'm hoping to delete this soon, so we could try to accelerate that if it's hard to fix.

File: Firebase/InstanceID/FIRInstanceIDTokenManager.h [hunk 2]
```diff
@@ -55,7 +54,7 @@ typedef NS_OPTIONS(NSUInteger, FIRInstanceIDInvalidTokenReason) {
  */
 - (void)fetchNewTokenWithAuthorizedEntity:(NSString *)authorizedEntity
                                     scope:(NSString *)scope
-                                  keyPair:(FIRInstanceIDKeyPair *)keyPair
+                                      IID:(NSString *)instanceID
```
Comment (chliangGoogle) [inline Firebase/InstanceID/FIRInstanceIDTokenManager.h]: maybe instanceID:(NSString *)instanceID to fit better on the style guide.

Comment (maksymmalyhin) [reply Firebase/InstanceID/FIRInstanceIDTokenManager.h]: Updated here and at other places

File: Firebase/InstanceID/FIRInstanceID.m [hunk 3]
```diff
@@ -779,17 +717,17 @@ - (void)setupTokenManager {
...
-  if ([self.keyPairStore invalidateKeyPairsIfNeeded]) {
-    // Reset tokens right away when keypair is deleted, otherwise async call can make first query
-    // of token happens before reset old tokens during app start.
-    // TODO(chliangGoogle): Delete all tokens on server too, using
-    // deleteAllTokensWithKeyPair:handler:. This requires actually retrieving the invalid keypair
-    // from Keychain, which is something that the key pair store does not currently do.
-    [self.tokenManager deleteAllTokensLocallyWithHandler:nil];
-  }
-}
+//- (void)setupKeyPairManager {
```
[the remark this answers is not in the document]
Comment (maksymmalyhin) [reply Firebase/InstanceID/FIRInstanceID.m]: deleted

**Pull request merged**